packages/sdk/src/controllers/ConnectorController.ts

```diff
@@ -189,6 +189,47 @@ export class ConnectorController {
         return res.getConnectorOptions(id).then((result) => getEditorResponseData<ConnectorOptions>(result));
     };
 
+    /**
+     * Sets the data source by id.
+     * @param connectorId the id of your data connector
+     * @returns
+     */
+    setDataSource = async (connectorId: string) => {
+        const res = await this.#editorAPI;
+        return res.setDataSource(connectorId).then((result) => getEditorResponseData<null>(result));
+    };
+
+    /**
+     * Gets the data source.
+     * @returns the id of your data connector
+     */
+    getDataSource = async () => {
+        // add backwards compatibility code
+        const res = await this.#editorAPI;
+        return res
+            .getDataSource()
+            .then((result) => getEditorResponseData<Next.ConnectorInstance>(result))
+            .then((resp) => {
+                const update: EditorResponse<ConnectorInstance> = { ...resp, parsedData: null };
+                if (resp.parsedData) {
+                    update.parsedData = this.#connectorCompatibilityTools.makeSingleConnectorBackwardsCompatible(
+                        resp.parsedData,
+                        this.#localConfig.get(WellKnownConfigurationKeys.GraFxStudioEnvironmentApiUrl),
+                    );
+                }
+                return update;
+            });

```

Comment:
DOes it need to be backwards compatible? Considering it is a new feature.

Reply:
I admit I did not know as I wanted to be safe there; I now just checked the compatibility source and indeed, shouldn't be a problem to remove it.